mlrun/execution.py

```diff
@@ -855,22 +862,27 @@ def commit(self, message: str = "", completed=True):
         if self._parent:
             self._parent.update_child_iterations()
             self._parent._last_update = now_date()
-            self._parent._update_db(commit=True, message=message)
+            self._parent._update_run(commit=True, message=message)
 
         if self._children:
             self.update_child_iterations(commit_children=True, completed=completed)
         self._last_update = now_date()
-        self._update_db(commit=True, message=message)
+        self._update_run(commit=True, message=message)
         if completed and not self.iteration:
             mlrun.runtimes.utils.global_context.set(None)
 
     def set_state(self, state: str = None, error: str = None, commit=True):
-        """modify and store the run state or mark an error
+        """
+        Modify and store the run state or mark an error
+        This method allows to set the run state to completed in the DB which is discouraged.
+        Completion of runs should be decided in the server.
 
-        :param state:   set run state
+        :param state:   set execution state
```

Comment:
why? you are using the context to update the run state

Reply:
the point is to separate the execution state from the run state.
i.e. if an execution state is completed it should not affect the run state but if an execution fails you can say that the run failed.
so yeah it affects the run state but they are not always correlated so it's not correct to say this it the run state.